packages/labs/context/src/lib/controllers/context-container.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ContextCallback} from '../context-request-event.js';
+
+/**
+ * A disposer function
+ */
+type Disposer = () => void;
+
+/**
+ * A tuple of context callback, and its disposer
+ */
+type CallbackRecord<T> = [ContextCallback<T>, Disposer];
+
+/**
+ * A simple class which stores a value, and triggers registered callbacks when the
+ * value is changed via its setter.
+ *
+ * An implementor might use other observable patterns such as MobX or Redux to get
+ * behavior like this. But this is a pretty minimal approach that will likely work
+ * for a number of use cases.
+ */
+export class ContextContainer<T> {
+  private callbacks: Set<CallbackRecord<T>> = new Set();
+
+  private _value!: T;
+  public get value(): T {
+    return this._value;
+  }
+  public set value(v: T) {
+    this.setValue(v);
+  }
+
+  public setValue(v: T, force = false) {
+    let changed = false;
+    if (v !== this._value) {
```

Comment:
Consider using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is.

Reply:
Good catch! Updated in 0248a2c